workflows/argo/prognostic_run_diags.yaml

```diff
@@ -6,76 +6,62 @@ spec:
   arguments:
     parameters:
     - name: runs
-    - name: make-movies
+    - name: recompute-diagnostics
       value: "false"
     - name: flags
       value: " "
-    - name: memory-compute-diags
-      value: "12Gi"
-    - name: memory-movie
-      value: "8Gi"
-    - name: memory-report
-      value: "6Gi"
   entrypoint: all
   volumes:
   - name: gcp-key-secret
     secret:
       secretName: gcp-key
   templates:
   - name: all
-    dag:
-      tasks:
-      - name: diagnostics-step
+    steps:
+    - - name: diagnostics-step
+        when: "{{workflow.parameters.recompute-diagnostics}} == true"
         withParam: "{{workflow.parameters.runs}}"
         template: diagnostics
         arguments:
           parameters:
             - name: run
               value: "{{item.url}}"
             - name: output
-              value: "gs://vcm-ml-public/argo/{{workflow.name}}/{{item.name}}"
+              value: "{{item.url}}_diagnostics"
             - name: flags
               value: "{{workflow.parameters.flags}}"
-            - name: memory
-              value: "{{workflow.parameters.memory-compute-diags}}"
-      - name: movie-step
-        when: "{{workflow.parameters.make-movies}} == true"
-        withParam: "{{workflow.parameters.runs}}"
-        template: movie
-        continueOn:
-          failed: true
-        arguments:
-          parameters:
-            - name: run
-              value: "{{item.url}}"
-            - name: output
-              value: "gs://vcm-ml-public/argo/{{workflow.name}}/{{item.name}}"
-            - name: memory
-              value: "{{workflow.parameters.memory-movie}}"
-      - name: generate-report
+    - - name: generate-report
         template: report
-        dependencies: [diagnostics-step, movie-step]
         arguments:
           parameters:
-            - name: memory
-              value: "{{workflow.parameters.memory-report}}"
+            - name: runs
+              value: "{{workflow.parameters.runs}}"
   - name: report
     inputs:
       parameters:
-      - name: memory
-        value: "6Gi"
+      - name: runs
     tolerations:
     - key: "dedicated"
       operator: "Equal"
       value: "med-sim-pool"
       effect: "NoSchedule"
     container:
       image: us.gcr.io/vcm-ml/fv3net
-      command:
-      - prognostic_run_diags 
-      - report
-      - gs://vcm-ml-public/argo/{{workflow.name}}
-      - gs://vcm-ml-public/argo/{{workflow.name}}
+      resources:
+        requests:
+          memory: "6Gi"
+          cpu: "1000m"
+        limits:
+          memory: "8Gi"
+          cpu: "1000m"
+      command: ["bash", "-c", "-x", "-e"]
+      args:
+      - |
+        cat << EOFRUNDIRSJSON > rundirs.json
+        {{inputs.parameters.runs}}
+        EOFRUNDIRSJSON
+        jq '.[].url |= . + "_diagnostics"' rundirs.json > diagnostic_dirs.json
```

Comment:
Would prefer to see this logic embedded in the `report-from-json` CLI maybe as a flag `--url-is-rundir` or something. This will make it easier to use the same rundirs.json locally and in argo.